torch/fx/symbolic_trace.py

```diff
@@ -660,10 +686,43 @@ def symbolic_trace(root : Union[torch.nn.Module, Callable], concrete_args: Optio
     Given an ``nn.Module`` or function instance ``root``, this function will return a ``GraphModule``
     constructed by recording operations seen while tracing through ``root``.
 
+    ``concrete_args`` allows you to partially specialize your function, whether it's to remove control flow or data structures.
+
+    For example::
+
+        def f(a, b):
+            if b == True:
+                return a
+            else:
+                return a*2
+
+    FX can typically not trace through this due to the presence of control
+    flow. However, we can use `concrete_args` to specialize on the value of
+    `b` to trace through this.
+
+        f = fx.symbolic_trace(f, concrete_args={'b': False})
+        assert f(3, False)  == 6
+
+    Note that although you can still pass in different values of `b`, they will be ignored.
+
+    We can also use `concrete_args` to eliminate data-structure handling from
+    our function. This will use pytrees to flatten your input. To avoid
+    overspecializing, pass in `fx.PH` for values that shouldn't be
+    specialized. For example::
+
+        def f(x):
+            out = 0
+            for v in x.values():
+                out += v
+            return out
```

Comment:
This PR adds pytree support for inputs. Is it a problem if a user returns a pytree instead of a Tensor? (Is that something that will be handled in the future or is that not something to worry about?)

Reply:
I have a test for this that's currently commented out: https://github.com/pytorch/pytorch/pull/55888/files#diff-75267f774cc7dff166215ce38fbbd50203da135346e52293386bb0697970a645R2363

Basically, FX already natively supports returning dicts/lists/tuples, but it doesn't support returning custom pytrees. So it's not *as* necessary. That's not something I've currently added, but will think about in the future.